test/com/amazon/ion/impl/lite/IonSystemLiteTest.java

```diff
@@ -0,0 +1,33 @@
+package com.amazon.ion.impl.lite;
+
+import com.amazon.ion.IonSystem;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.system.IonSystemBuilder;
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class IonSystemLiteTest {
+    // Confirms that the standard IonSystem can materialize Ion data that is deeply nested without
+    // exhausting the stack.
+    @Test
+    void materializeDeeplyNested() {
```

Comment:
🗺️ This unit test fails due to stack exhaustion when using the code in `origin/master`:

![image](https://github.com/amazon-ion/ion-java/assets/611616/8e218acb-ecc3-4e3c-8186-9153ad719824)